### ISSUE TITLE
error evaluating kustomization manifest for dex Error

### ISSUE THREAD

**Issue by bharath0208** [kubeflow/manifests#1363]
Hi All,

I am getting this error when trying to apply this file` kfctl apply -f kfctl_istio_dex.yaml `

```ERRO[0020] error evaluating kustomization manifest for dex Error field specified in var '{namespace ~G_v1_ConfigMap {data.namespace}}' not found in corresponding resource  filename="kustomize/kustomize.go:175"
Error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: error evaluating kustomization manifest for dex Error field specified in var '{namespace ~G_v1_ConfigMap {data.namespace}}' not found in corresponding resource
Usage:
  kfctl apply -f ${CONFIG} [flags]

Flags:
  -f, --file string   Static config file to use. Can be either a local path:
                      		export CONFIG=./kfctl_gcp_iap.yaml
                      	or a URL:
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_gcp_iap.v1.0.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_istio_dex.v1.0.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_aws.v1.0.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_k8s_istio.v1.0.0.yaml
                      	kfctl apply -V --file=${CONFIG}
  -h, --help          help for apply
  -V, --verbose       verbose output default is false

failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: error evaluating kustomization manifest for dex Error field specified in var '{namespace ~G_v1_ConfigMap {data.namespace}}' not found in corresponding resource```

**Comment by amybachir** [comment]
I ran into the same issue today. I have to say the installation was not smooth at all. I had to mess with a bunch of kustomize files to get it working. Here is how I got through the "data.namespace" issue.

1. remove the namespace variable reference completely from this file kustomize/dex/kustomization.yaml
2. add the following in kustomize/dex/base/kustomization.yaml
```
- name: namespace
  objref:
    kind: ConfigMap
    name: dex-parameters
    apiVersion: v1
  fieldref:
    fieldpath: data.namespace

```

1. add the following variable in kustomize/dex/base/params.env
`namespace=auth`

You are going to run into more issues. A lot more issues...

**Comment by evonit** [comment]
In my case, simply append one line in kustomize/dex/base/params.env
namespace=auth
It works. got another error though...

**Comment by Jeffwan** [comment]
See #1424

**Comment by PatrickXYS** [comment]
> I am getting this error when trying to apply this filekfctl apply -f kfctl_istio_dex.yaml

I would suggest using stable version manifest, such as `kfctl_istio_dex.v1.0.2.yaml`, master version of manifest synced with master branch and might caused some unpredictable issue.

**Comment by DavidSpek** [comment]
After setting the dex namespace as metioned above the following error is related to webhook. It seems like webhook has not yet been migrated to v3. I tried setting `path: admission-webhook/webhook/v3` in kfctl_istio_dex.yaml which did not help. After that I tried editing `admission-webhook/webhook/overlays/cert-manager/kustomization.yaml` and changing 
```
bases:
- ../../base
```

to 
```
bases:
- ../../v3
```
Sadly this has not resolved the issue, but I thought this information might help others looking for a solution.
